pkg/test/framework/components/echo/kube/deployment.go

```diff
@@ -173,19 +173,37 @@ func generateYAML(cfg echo.Config) (string, error) {
 	if err != nil {
 		return "", err
 	}
+	return generateYAMLWithSettings(cfg, settings)
+}
+
+func generateYAMLWithSettings(cfg echo.Config, settings *image.Settings) (string, error) {
+	// Convert legacy config to workload oritended.
+	if cfg.Subsets == nil {
+		cfg.Subsets = []echo.SubsetConfig{
+			{
+				Version: cfg.Version,
+			},
+		}
+	}
 
 	// Separate the annotations.
 	serviceAnnotations := make(map[string]string)
-	workloadAnnotations := make(map[string]string)
-	for k, v := range cfg.Annotations {
-		switch k.Type {
-		case echo.ServiceAnnotation:
-			serviceAnnotations[k.Name] = v.Value
-		case echo.WorkloadAnnotation:
-			workloadAnnotations[k.Name] = v.Value
-		default:
-			scopes.Framework.Warnf("annotation %s with unknown type %s", k.Name, k.Type)
+	wlas := make(map[string]map[string]string)
```

Comment:
Why not just modify the annotations in the subset directly? Then you can just reference the annotations for the subset directly in the yaml, rather than using yet another data structure for annotations.

Reply:
I'm following the existing code, see `serviceAnnotations` and previous `workloadAnnoations`.

The type of annotations is different from map[string]string, its to a struct with .Values as string. If i remembered correctly, referencing them directly in gotmpl causes problems. (i guess that's why we do conversion for serviceAnnoations as well)

Reply:
hmm ... could you give it a try to confirm that it's a problem?

Reply:
Done. Actually we can reference the struct in gotmpl.thus the existing svc annot is not needed as well. i make that consistent as well.